packages/server/src/adapters/next-app-dir/nextAppDirCaller.ts

```diff
@@ -19,9 +19,13 @@ import { rethrowNextErrors } from './rethrowNextErrors';
 /**
  * Create a caller that works with Next.js React Server Components & Server Actions
  */
-export function nextAppDirCaller<TContext>(
+export function nextAppDirCaller<TContext, TMeta>(
   config: Simplify<
     {
+      /**
+       * Extract the path from the procedure metadata
+       */
+      pathExtractor?: (meta: TMeta) => string;
```

Comment:
options object allow addition of more things in the future without breaking the API
```suggestion
      pathExtractor?: (opts: { meta: TMeta }) => string;
```